job_bundles/Using_Files_In_Jobs.md

```diff
@@ -0,0 +1,1145 @@
+and then a second for the task in the step named "DependentStep".
+
+To see how this works, first start the Deadline Cloud worker agent in an AWS CloudShell tab as you did when following
+along in the section on [Getting Output Files from a Job](#32-getting-output-files-from-a-job).
+Let all of the previously submitted jobs finish running if there are any, and then delete the job logs from the logs directory:
+
+```bash
+rm -rf ~/devdemo-logs/queue-*
+```
+
+Then, submit a job using the modified `job_attachments_devguide_output` job bundle, wait for it to finish running on the
+worker in your AWS CloudShell environment, and then look out the logs for the two sessions that are run:
+
+```bash
+# Change the value of FARM_ID to your farm's identifier
+FARM_ID=farm-00112233445566778899aabbccddeeff
+# Change the value of QUEUE1_ID to queue Q1's identifier
+QUEUE1_ID=queue-00112233445566778899aabbccddeeff
+
+deadline bundle submit --farm-id $FARM_ID --queue-id $QUEUE1_ID ./job_attachments_devguide_output
+
+# Wait for the job to finish running, and then:
+
+cat demoenv-logs/queue-*/session-*
+```
+
+In the session log for the task in the step named `DependentStep`, notice that there are two separate
+download actions that are run:
+
+```
+2024-07-17 02:52:05,666 INFO ==============================================
+2024-07-17 02:52:05,666 INFO --------- Job Attachments Download for Job
+2024-07-17 02:52:05,667 INFO ==============================================
+2024-07-17 02:52:05,667 INFO Syncing inputs using Job Attachments
+2024-07-17 02:52:05,928 INFO Downloaded 207.0 B / 207.0 B of 1 file (Transfer rate: 0.0 B/s)
+2024-07-17 02:52:05,929 INFO Summary Statistics for file downloads:
+Processed 1 file totaling 207.0 B.
+Skipped re-processing 0 files totaling 0.0 B.
+Total processing time of 0.03954 seconds at 5.23 KB/s.
+
+2024-07-17 02:52:05,979 INFO 
+2024-07-17 02:52:05,979 INFO ==============================================
+2024-07-17 02:52:05,979 INFO --------- Job Attachments Download for Step
+2024-07-17 02:52:05,979 INFO ==============================================
+2024-07-17 02:52:05,980 INFO Syncing inputs using Job Attachments
+2024-07-17 02:52:06,133 INFO Downloaded 117.0 B / 117.0 B of 1 file (Transfer rate: 0.0 B/s)
+2024-07-17 02:52:06,134 INFO Summary Statistics for file downloads:
+Processed 1 file totaling 117.0 B.
+Skipped re-processing 0 files totaling 0.0 B.
+Total processing time of 0.03227 seconds at 3.62 KB/s.
+```
+
+The first of these is downloading the `script.sh` file that is run by the step named "Step", and the second
+is downloading the outputs from that step. Then later in that same log, you can see the output from the step
+named "DependentStep":
+
+```
+2024-07-17 02:52:06,213 INFO Output:
+2024-07-17 02:52:06,216 INFO Script location: /sessions/session-5b33fa6d5ea94861ba4cd88a9e8d31d94kszcb0j/assetroot-e7d860e9a7a86e6bbf77/script.sh
+```
```

Comment:
Super nit, do we want to do an aws s3 ls --recursive s3://$Q1_S3_BUCKET/DeadlineCloud/Manifests/$JOB_ID to list out the output manifests executed by the first step? This will give the reader more context how the step-step data is exchanged. A step-step dependency is no different than an input Job Attachment. The output manifest of prior steps become one of the manifests to download for dependent jobs.

Reply:
I'm not sure. Do you think that we need to make that explicit? The output manifest is the same as explained in the section talking about job outputs. I can add a sentence here, though, to make it clear that the dependent step's "Download for Step" is using the output manifest from "Step" as an input manifest.